app/services/process_steps.rb

```diff
@@ -0,0 +1,40 @@
+class ProcessSteps
+  def initialize(steps:, step:, adjust:)
+    @steps = steps
+    @step = step
+    @adjust = adjust
+  end
+
+  def current_step_number
+    current_step = if defined?(@step)
+                     @step
+                   else
+                     params[:controller].split("/").last.to_sym == :documents ? :documents : :review
+                   end
+    @steps[current_step][:number] - @adjust
+  end
+
+  def steps_to_display
+    steps = @steps.dup
+    if @adjust.positive?
```

Comment:
Not sure if I follow completely – the steps below seem to be specific to _one_ particular process, but this service looks like it should be generic 😕